src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -95,9 +95,10 @@ def is_valid_connection_string(connection_string):
         segments = connection_string.split(';')
         if len(segments) != 3:
             return False
-        if segments[0][:9] != 'Endpoint=' or segments[1][:3] != 'Id=' or segments[2][:7] != 'Secret=':
-            return False
-        return True
+
+        segments.sort()
+        if segments[0].startswith('Endpoint=') and segments[1].startswith('Id=') and segments[2].startswith('Secret='):
```

Comment:
>'Endpoint=' [](start = 34, length = 11)

This has always worked well in the past, but to be on the safer side - do you know if connection strings are case sensitive? I wasn't able to find any concrete documentation about connection string formats.

Reply:
Yeah, there is no hard requirement for the format of connection string. In most case, clients should get it from either CLI or portal and use it directly. However, we saw a customer reported issue in GitHub Action complaining for the order of those three segments. In CLI, didn't hear any complains about the format either order or case sensitivity. So I guess we leave it as case sensitive until there is a real ask. 

---
In reply to: [456233992](https://github.com/Azure/azure-cli/pull/14398#discussion_r456233992) [](ancestors = 456233992)

Reply:
Sure, makes sense!

---
In reply to: [456708582](https://github.com/Azure/azure-cli/pull/14398#discussion_r456708582) [](ancestors = 456708582,456233992)